Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/frame/DartVmServiceValue.java

```diff
@@ -149,10 +149,42 @@ public void computePresentation(@NotNull final XValueNode node, @NotNull final X
     if (computeRegExpPresentation(node)) return;
     if (computeMapPresentation(node)) return;
     if (computeListPresentation(node)) return;
-    computeDefaultPresentation(node);
+
+    // computeDefaultPresentation is called internally here when no result is got.
+    // The reason for this is that the async method used cannot be properly waited.
+    computeDebugPresentation(node);
+
     // todo handle other special kinds: Type, TypeParameter, Pattern, may be some others as well
   }
 
+  private void computeDebugPresentation(final XValueNode node) {
+    myDebugProcess.getVmServiceWrapper()
+      .evaluateInTargetContext(myIsolateId, myInstanceRef.getId(), "toStringDeep()", new VmServiceConsumers.EvaluateConsumerWrapper() {
+        @Override
+        public void received(final InstanceRef toStringInstanceRef) {
+          if (toStringInstanceRef.getKind() == InstanceKind.String) {
+            String content = toStringInstanceRef.getValueAsString();
+            String summary = content.split("\n")[0];
+            node.setPresentation(getIcon(), myInstanceRef.getClassRef().getName(), summary, true);
+            if (toStringInstanceRef.getValueAsStringIsTruncated()) {
+              addFullStringValueEvaluator(node, toStringInstanceRef);
+            }
+            else {
+              node.setFullValueEvaluator(new ImmediateFullValueEvaluator("View", content));
```

Comment:
Probably we don't need 'View' link if `content` is single-line?

Reply:
Fixed. Thanks!

Reply:
I mean, when `content` is just one line then it is equal to `summary` and `summary` is already shown, so in such case, we probably don't need `View` link at all.

Reply:
Yes, I get it. Take a look at the file after all three commits.

Reply:
The view link is only added when it's multi-line now.

Reply:
Correct! Sorry, I looked at a wrong commit.